src/BitStringAddresses/bosefs.jl

```diff
@@ -327,36 +325,60 @@ end
         result += bosons
         chunk >>>= bosons % UInt
     end
-    return (result, orbital, bit_position), (i, chunk, bits_left, orbital)
+    return BoseFSIndex(result, orbital, bit_position), (i, chunk, bits_left, orbital)
 end
 
-"""
-    BoseFS2C{NA,NB,M,AA,AB} <: AbstractFockAddress
+function find_site(b::BoseFS, index)
+    last_occnum = last_site = last_offset = 0
+    for (occnum, site, offset) in occupied_orbitals(b)
+        dist = index - site
+        if dist == 0
+            return BoseFSIndex(occnum, index, offset)
+        elseif dist < 0
+            return BoseFSIndex(0, index, offset + dist)
+        end
+        last_occnum = occnum
+        last_site = site
+        last_offset = offset
+    end
+    offset = last_offset + last_occnum + index - last_site
+    return BoseFSIndex(0, index, offset)
+end
 
-Address type that constructed with two [`BoseFS{N,M,S}`](@ref). It represents a
-Fock state with two components, e.g. two different species of bosons with particle
-number `NA` from species S and particle number `NB` from species B. The number of
-orbitals `M` is expected to be the same for both components.
-"""
-struct BoseFS2C{NA,NB,M,SA,SB} <: AbstractFockAddress
-    bsa::BoseFS{NA,M,SA}
-    bsb::BoseFS{NB,M,SB}
+function find_particle(b::BoseFS, index, n=1)
+    for (occnum, site, offset) in occupied_orbitals(b)
+        index -= occnum ≥ n
+        if index == 0
+            return BoseFSIndex(occnum, site, offset)
+        end
+    end
+    return BoseFSIndex(0, 0, 0)
 end
 
-BoseFS2C(onr_a::Tuple, onr_b::Tuple) = BoseFS2C(BoseFS(onr_a),BoseFS(onr_b))
+"""
+    move_particle(b::BoseFS, from::BoseFSIndex, to::BoseFSIndex)
 
-function Base.show(io::IO, b::BoseFS2C{NA,NB,M,AA,AB}) where {NA,NB,M,AA,AB}
-    print(io, "BoseFS2C(")
-    Base.show(io,b.bsa)
-    print(io, ",")
-    Base.show(io,b.bsb)
-    print(io, ")")
-end
+Move particle from [`BoseFSIndex`](@info) `from` to the [`BoseFSIndex`](@ref) `to`.
 
-num_particles(::Type{<:BoseFS2C{NA,NB}}) where {NA,NB} = NA + NB
-num_modes(::Type{<:BoseFS2C{<:Any,<:Any,M}}) where {M} = M
-Base.isless(a::BoseFS2C, b::BoseFS2C) = isless((a.bsa, a.bsb), (b.bsa, b.bsb))
+This is equivalent to applying a destruction operator followed by a creation operator to the
+address.
 
-function nearUniform(::Type{<:BoseFS2C{NA,NB,M}}) where {NA,NB,M}
-    return BoseFS2C(nearUniform(BoseFS{NA,M}), nearUniform(BoseFS{NB,M}))
+Return the new fock state and the product of the occupation numbers.
```

Comment:
```suggestion
Return the new Fock state and the product of the occupation numbers.
```

I mentioned in `fermifs.jl` that it would be cool to have this function carry out exactly the same operation as the creation and destruction operators. The correct amplitude then would be the square root of what we are returning now. In early versions of the code I decided to return the integer product (as is currently done) and take the square root later when converting to `Float64`. I'm not sure whether it would make any performance difference.